site/models/plugin.php

```diff
@@ -151,7 +174,29 @@ public function toJson($onlyIfCached = false)
             'category'    => option('plugins.categories.' . $this->category() . '.label'),
             'description' => $this->description()->value(),
             'screenshot'  => $screenshot ? $screenshot->url() : null,
-            'version'     => $this->version($onlyIfCached)
+
+            // basic skeleton for the update check (can be extended later)
+            'latest'   => $version,
+            'versions' => [
+                $version => [
+                    'description' => 'Latest release',
+                    'status'      => 'latest'
+                ],
+                '*' => [
+                    'description' => 'Actively supported',
+                    'latest'      => $version,
+                    'status'      => 'active-support'
+                ]
+            ],
+            'urls' => [
+                '*' => [
+                    'changes'  => $this->changes('$version'),
+                    'download' => $this->download('$version'),
```

Comment:
I think these two are not supposed to be wrapped as strings?

Reply:
They are. `$version` is a placeholder in the URL template.

Reply:
Why are we doing it like this and not passing the actual version here? This. is very confusing to read tbh - especially sine `::changes()` apparently is also supposed to be called with an actual version number?

Reply:
The update checking code can use the URL template to build URLs for any version, not just the latest one. This is e.g. necessary in the following cases:

- If there is a paid upgrade but the update check wants to suggest the last free update.
- If there are vulnerabilities and the update check wants to suggest the minimum possible security update.

I've added a comment for future reference.
The format of the security data is also documented here: https://github.com/getkirby/getkirby.com/blob/de11bdecb1542cfccba3314ac994baafba8e9efa/content/security/security.txt#L38

Reply:
I would suggest using `'{{ version }}'` as that's our common string template placeholder, that also works directly with `$helper.string.template()`.

Reply:
I agree about the placeholder for version. The rest looks good to me.

Reply:
@distantnative @bastianallgeier I tried it, but frankly it's becoming a mess. We already use the placeholder syntax in our security data so that getkirby.com can fill some information dynamically:

https://github.com/getkirby/getkirby.com/blob/b7b296e11b105190ba71cb9d7a5028597aaf8db7/content/security/security.txt#L179-L191

If we also use the same syntax for the placeholders that should be kept, it's going to be really confusing IMO.

Reply:
Why would it be a mess? For me the referenced lines would be rather an argument to make it consistent. Right now we have to different ways of placeholders in the same place - to me that's the really confusing part. One consistent style (and I think there is a good argument for `{{ }}` und the Kirby world) would be much better.

Reply:
`Str::template()` also supports only partial replacement. So getkirby.com can already replace what it needs to replace and the Panel fills in the remaining placeholder.

Reply:
@lukasbestle I guess your template PR is exactly about solving this problem, right?

Reply:
> @lukasbestle I guess your template PR is exactly about solving this problem, right? 

Yes, exactly.

Reply:
I just find it confusing that we'll have some placeholders that are replaced by getkirby.com and some placeholders that are left over for dynamic replacement. Down the line we won't know which are which without looking it up.

I'm fine with the change, but then I'd suggest to use a custom format for the "static" placeholders, e.g. `[{ latestMajor }]`. What do you think?

Reply:
But now that I think about it: Does the `$field->replace()` method even support custom placeholder start and end strings?

Reply:
That would be in the content file. Since it's YAML, this could work. I'll try it.

Reply:
Or even in the system class where it then gets processed.

Reply:
I think there is a misunderstanding. This is all just about getkirby.com. The CMS implementation is no problem at all.

Reply:
That's evident. But I don't think it would be a huge problem if there is not comment in the content file. If there is a comment in the PHP code that replaces the first placeholder, that's fine. If there is just a comment on the side of Kirby that actually interprets the input, that'd be totally fine as well. Those placeholders are quite self-explanatory as placeholders if we stick to our code style. And then it's actually not so hard to follow where what gets replaced.